### PR TITLE
Feature/user pantry view

### LINKED ISSUE #15: User Stories Part III, scripts.js
- [x] As a user, I should be able to view what ingredients exist inside of my pantry.
- [x] As a user, I should be able to check my list of recipes to cook and see if my pantry has enough ingredients to cook a meal.
- [x] As a user, I should be told what ingredients are still needed if I don’t have enough ingredients in my pantry to cook the recipe.
- [x] As a user, I should be able to cook a meal if I do have enough ingredients in my pantry. As a result, cooking the meal should remove the ingredients from my pantry.

### LINKED ISSUE #51: DOM manipulation: User Story: As a user I should be able to add a recipe to a list of recipes to cook
- [x]  When the My Pantry button is clicked, The user should see an area that hold's their recipes they added to a cook list

- Will this be added when the Add to cook button is clicked? or when the My pantry button is clicked?

### LINKED ISSUE #57: DOM manipulation: User should be able to view a list of ingredients in their pantry
- [x] When the My pantry button is clicked, the user should be taken to a new display and their pantry items should be displayed with amount
- [x] When the My pantry button is clicked, the title on the page should change to My pantry

### LINKED ISSUE #59: DOM Manipulation: As a User I should be told what ingredients are still needed if I don't have enough ingredients in my Pantry
- [x] When the Cook This button is clicked, a list of ingredients that are needed should populate in the box below and a message should appear " You need to buy these ingredients"

### LINKED ISSUE #60: DOM manipulation: User story: As a user i should be be able to cook a meal if I DO have enough ingredients in my pantry and ingredients should be removed from the pantry
- [x] If The user has all the ingredients, a message should display, "You have updated your pantry, You have cooked this recipe..etc"

### PR DESCRIPTION
**Pull request by alexthompson207**
## Is this a fix or a feature?

This is a feature.

## What is the change?

- As a user, I am able to view what ingredients exist inside of my pantry.
- As a user, I am able to check my list of recipes to cook and see if my pantry has enough ingredients to cook a meal.
- As a user, I am told what ingredients are still needed if I don’t have enough ingredients in my pantry to cook the recipe.
- As a user, I am able to cook a meal if I do have enough ingredients in my pantry. As a result, cooking the meal removes the ingredients from my pantry.

## What does it fix?

N/a.

## Where should the reviewer start?

In the `scripts.js` file:
Start at line 223: `handleCookThisButton()`;
Line 237: `evaluatePantry(recipe)`;
Line 252: `displayMissingIngredients(ingredients)`

## How should this be tested?

As a user you should be able to view a recipe, and click on the `add to cook` button, once recipes are added to via the `add to cook` button, clicking on the `my pantry` tab should bring the user to a pantry view. From the pantry view a user can view their pantry ingredients and amounts. Clicking on a meal in the `Meals to Cook` box, a users pantry ingredients and amounts should be evaluated on the recipe ingredients and amounts they are trying to cook, as a result 2 things could happen. If a user has enough ingredients to cook a recipe, their pantry will be updated to reflect using those ingredients to make the recipe. If a user doesn't have enough ingredients a message will display letting the user know that don't have enough ingredients, and the missing ingredients needed to cook a recipe will be listed with the amounts needed.

## Screenshots (if appropriate):

screencapture-file-Users-alexthompson-turing-mod-2-projects-whats-cookin-src-index-html-2021-02-15-16_49_55

### CODE REVIEW

**Comment by alexthompson207** [comment]
Things to refactor:
- [x] Pantry styling needs to extend down if pantry ingredients are larger than the container
- [x] What happens when a user clicks on `Cook This` button but hasn't selected a recipe, or no recipes have been added to meals to cook
- [x] Maybe add cursor pointer when hovering over buttons?
- [x] Favorites view- what happens when user has no favorites, do we want to display anything?

File: src/scripts.js [hunk 1]
```diff
@@ -200,32 +208,58 @@ function handleSingleRecipeButtons(event) {
...
+function handleCookThisButton() {
+  let recipeSelection;
+  const cookList = document.querySelectorAll(".pantry-btn")
+  cookList.forEach(recipe => {
+    if (recipe.checked) {
+      recipeSelection = recipesRepo.filterRecipesByName(recipe.value);
+    } else {
+      //What should happen here
+      console.log("Please make a selection");
+    }
```
Comment (elizhahn) [inline src/scripts.js:232]: I was thinking here, if none of the inputs are checked, we could put some code here to manipulate the DOM to display a message to the user...like "Please select a recipe to cook"

File: src/scripts.js [hunk 2]
```diff
@@ -248,13 +283,43 @@ function handleNavButtons(event) {
...
+    <p class="single-recipe-number">${ingredient.amount}</p>
+    <p class="single-recipe-ingredient">${ingredient.name}</p>
+     </li>`
+  })
+}
+
+function findPantryIngredientNames() {
+  currentUser.pantry.pantry.map(ingredient => {
+    return ingredient.name =
+      ingredientRepo.returnIngredientName(ingredient.ingredient)
```
Comment (elizhahn) [inline src/scripts.js]: Possibly a rename here, since this is adding ingredient names to a user's pantry. Maybe we call it, addIngredientNames?

**Comment by elizhahn** [comment]
Other considerations
- [x] when a pantry amount is 0 do we want to remove it from the pantry list?

File: src/scripts.js (hunk 1 above)
Comment (alexthompson207) [reply src/scripts.js:232]: Yeah I was thinking the same thing. I like it!

File: src/scripts.js (hunk 2 above)
Comment (alexthompson207) [reply src/scripts.js]: Yeah I like it! Updated the function name, and pushed it up!

**Pull request merged**